core/domain/subtopic_page_services_test.py

```diff
@@ -99,14 +107,53 @@ def test_get_subtopic_page_contents_by_id(self):
                 'property_name': 'page_contents_html',
                 'new_value': 'a',
                 'old_value': 'b'
-            })]
-        )
+            })])
         subtopic_page_contents = (
             subtopic_page_services.get_subtopic_page_contents_by_id(
                 self.TOPIC_ID, 1))
         self.assertEqual(
             subtopic_page_contents.to_dict(), expected_page_contents_dict)
 
+    def test_save_subtopic_page(self):
+        subtopic_page_1 = (
+            subtopic_page_domain.SubtopicPage.create_default_subtopic_page(
+                1, 'topic_id_1'))
+        subtopic_page_services.save_subtopic_page(
+            self.user_id, subtopic_page_1, 'Added subtopic',
+            [topic_domain.TopicChange({
+                'cmd': topic_domain.CMD_ADD_SUBTOPIC,
+                'subtopic_id': 1,
+                'title': 'Sample'
+            })])
+        with self.assertRaises(Exception):
+            subtopic_page_services.save_subtopic_page(
+                self.user_id, subtopic_page_1, 'Added subtopic', [])
+        subtopic_page_id_1 = (
+            subtopic_page_domain.SubtopicPage.get_subtopic_page_id(
+                'topic_id_1', 1))
+        subtopic_page_model_1 = topic_models.SubtopicPageModel.get(
+            subtopic_page_id_1)
+        subtopic_page_1.version = 2
+        subtopic_page_model_1.version = 3
+        with self.assertRaises(Exception):
+            subtopic_page_services.save_subtopic_page(
+                self.user_id, subtopic_page_1, 'Added subtopic',
+                [topic_domain.TopicChange({
+                    'cmd': topic_domain.CMD_ADD_SUBTOPIC,
+                    'subtopic_id': 1,
+                    'title': 'Sample'
+                })])
+        subtopic_page_1.version = 3
+        subtopic_page_model_1.version = 2
+        with self.assertRaises(Exception):
```

Comment:
ditto

Reply:
Yeah, I used the raisesRegexp to check for the actual exception (as mentioned in the function) which is being raised.